src/io/deviceManager.js

```diff
@@ -312,7 +312,17 @@ class DeviceManager {
         };
         if (deviceSpec) queryObject.spec = deviceSpec;
         const url = `${this._serverURL}/${encodeURIComponent(deviceType)}/list?${querystring.stringify(queryObject)}`;
-        return got(url).then(response => JSON.parse(response.body));
+        return new Promise((resolve, reject) => {
+            nets({
+                method: 'GET',
+                url: url,
+                json: {}
+            }, (err, res, body) => {
+                if (err) return reject(err);
+                if (res.statusCode !== 200) reject(body);
```

Comment:
I recommend `return reject(body);` here, or switching it all to an `if`/`else` chain.

Reply:
Good catch. That is simply an oversight on my part.